### ISSUE TITLE
Override Compute Loss Function for MeZO Layer3

### ISSUE THREAD

**Issue by MustafaKhan0** [alansun17904/MAAA#27]
# Summary

Override the compute_loss function for the MeZOTrainer in the layer3/ codebase, such that it follows the equation specified in the research plan. Specifically, in src/layer3/l3mezo.py. 

Research plan:

We first construct a counterfacutal dataset: for each input $x$ and counterfactual input $x'$ we yield model predictions $M(x), M(x')$. Then, we finetune $M$ to create $M^\star$ using MeZO to minimize the following equation. 

Equation: $$\text{KL}(M(x') \Vert M^\star(x) ) + \lambda \lVert M - M^\star\rVert$$ 
where $\lambda > 0$ and $\lVert M - M^\star\rVert$ measures the change in weights. 

The branch for this issue should be created using the layer3 branch, and should be merged back into the layer3 branch. 

Another issue (currently being drafted) will take care of implementing a function which computes the change in weights. 

# Proposed Changes

See the comments in src/layer3/l3mezo.py for where to change the code. The loss should be entirely computed, using a function that which computes the difference between the weights. 

# Acceptance Criteria

The loss is correctly computed and implemented based on the formula above.

**Comment by MustafaKhan0** [comment]
Made the changes on the layer3 branch - will discuss and review during the next meeting, and test after #26 is complete!
